src/box/sql/func.c

```diff
@@ -1008,6 +1008,36 @@ func_round(struct sql_context *ctx, int argc, const struct Mem *argv)
 	return mem_set_double(res, (double)(int64_t)(d + delta));
 }
 
+/** Implementation of the ROUND() function for DECIMAL argument. */
+static void
+func_round_dec(struct sql_context *ctx, int argc, const struct Mem *argv)
+{
+	assert(argc == 1 || argc == 2);
+	if (mem_is_null(&argv[0]) || (argc == 2 && mem_is_null(&argv[1])))
+		return;
+	assert(mem_is_dec(&argv[0]));
+	assert(argc == 1 || mem_is_int(&argv[1]));
+	uint64_t n = (argc == 2 && mem_is_uint(&argv[1])) ? argv[1].u.u : 0;
+
+	if (n >= DECIMAL_MAX_DIGITS)
+		return mem_set_dec(ctx->pOut, &argv[0].u.d);
+	decimal_t dec = argv[0].u.d;
+	decimal_round(&dec, n);
+	return mem_set_dec(ctx->pOut, &dec);
```

Comment:
I suggest this diff to reduce some copying:
<details><summary>diff</summary>

```Diff
 	assert(mem_is_dec(&argv[0]));
 	assert(argc == 1 || mem_is_int(&argv[1]));
 	uint64_t n = (argc == 2 && mem_is_uint(&argv[1])) ? argv[1].u.u : 0;
-
-	if (n >= DECIMAL_MAX_DIGITS)
-		return mem_set_dec(ctx->pOut, &argv[0].u.d);
-	decimal_t dec = argv[0].u.d;
-	decimal_round(&dec, n);
-	return mem_set_dec(ctx->pOut, &dec);
+	mem_set_dec(ctx->pOut, &argv[0].u.d);
+	if (n < DECIMAL_MAX_DIGITS)
+		decimal_round(&ctx->pOut->u.d, n);
 }
```
</details>

Reply:
Thank you for the review and suggestion! Applied.